syft/pkg/cataloger/deb/parse_dpkg_status.go

```diff
@@ -59,9 +58,6 @@ func parseDpkgStatusEntry(reader *bufio.Reader) (pkg.DpkgMetadata, error) {
 	var retErr error
 	dpkgFields, err := extractAllFields(reader)
 	if err != nil {
-		if !errors.Is(err, errEndOfPackages) {
-			return pkg.DpkgMetadata{}, err
-		}
```

Comment:
looking below at `extractAllFields`, it appears to also only return fields,nil or nil,err; so continuing processing here also would be wrong, no?